runtime/lua/vim/lsp/buf.lua

```diff
@@ -64,20 +64,32 @@ function M.completion(context)
   return request('textDocument/completion', params)
 end
 
-function M.formatting(options)
+local function formatting_params(options)
   validate { options = {options, 't', true} }
   local sts = vim.bo.softtabstop;
   options = vim.tbl_extend('keep', options or {}, {
     tabSize = (sts > 0 and sts) or (sts < 0 and vim.bo.shiftwidth) or vim.bo.tabstop;
     insertSpaces = vim.bo.expandtab;
   })
-  local params = {
+  return {
     textDocument = { uri = vim.uri_from_bufnr(0) };
     options = options;
   }
+end
+
+function M.formatting(options)
+  local params = formatting_params(options)
   return request('textDocument/formatting', params)
 end
 
+function M.formatting_sync(options, timeout)
```

Comment:
Would you add this function to `runtime/doc/lsp.txt`?

Reply:
Good point, will do.